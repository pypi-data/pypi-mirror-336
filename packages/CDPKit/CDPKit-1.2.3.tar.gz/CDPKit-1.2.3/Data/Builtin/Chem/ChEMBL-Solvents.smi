WATER	[OH2]
DICHLOROMETHANE	ClCCl
TRICHLOROMETHANE	ClC(Cl)Cl
ETHYL ACETATE	CCOC(=O)C
METHANOL	CO
PROPAN-2-OL	CC(C)O
ACETONE	CC(=O)C
DMSO	CS(=O)C
ETHANOL	CCO
