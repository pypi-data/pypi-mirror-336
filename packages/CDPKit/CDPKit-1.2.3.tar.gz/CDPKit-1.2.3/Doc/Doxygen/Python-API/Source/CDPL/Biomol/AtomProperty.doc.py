#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Provides keys for built-in Biomol.Atom properties.
# 
class AtomProperty(Boost.Python.instance):

    ##
    # \brief 
    #
    RESIDUE_ATOM_NAME = CDPL.Base.LookupKey('RESIDUE_ATOM_NAME')

    ##
    # \brief 
    #
    RESIDUE_ALT_ATOM_NAME = CDPL.Base.LookupKey('RESIDUE_ALT_ATOM_NAME')

    ##
    # \brief 
    #
    RESIDUE_LEAVING_ATOM_FLAG = CDPL.Base.LookupKey('RESIDUE_LEAVING_ATOM_FLAG')

    ##
    # \brief 
    #
    RESIDUE_LINKING_ATOM_FLAG = CDPL.Base.LookupKey('RESIDUE_LINKING_ATOM_FLAG')

    ##
    # \brief 
    #
    RESIDUE_CODE = CDPL.Base.LookupKey('RESIDUE_CODE')

    ##
    # \brief 
    #
    RESIDUE_SEQUENCE_NUMBER = CDPL.Base.LookupKey('RESIDUE_SEQUENCE_NUMBER')

    ##
    # \brief 
    #
    RESIDUE_INSERTION_CODE = CDPL.Base.LookupKey('RESIDUE_INSERTION_CODE')

    ##
    # \brief 
    #
    HETERO_ATOM_FLAG = CDPL.Base.LookupKey('HETERO_ATOM_FLAG')

    ##
    # \brief 
    #
    CHAIN_ID = CDPL.Base.LookupKey('CHAIN_ID')

    ##
    # \brief 
    #
    ALT_LOCATION_ID = CDPL.Base.LookupKey('ALT_LOCATION_ID')

    ##
    # \brief 
    #
    ENTITY_ID = CDPL.Base.LookupKey('ENTITY_ID')

    ##
    # \brief 
    #
    MODEL_NUMBER = CDPL.Base.LookupKey('MODEL_NUMBER')

    ##
    # \brief 
    #
    SERIAL_NUMBER = CDPL.Base.LookupKey('SERIAL_NUMBER')

    ##
    # \brief 
    #
    OCCUPANCY = CDPL.Base.LookupKey('OCCUPANCY')

    ##
    # \brief 
    #
    B_FACTOR = CDPL.Base.LookupKey('B_FACTOR')
