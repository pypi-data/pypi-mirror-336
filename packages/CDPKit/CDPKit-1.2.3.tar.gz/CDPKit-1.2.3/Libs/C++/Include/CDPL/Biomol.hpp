/* 
 * Biomol.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::Biomol.
 */

#ifndef CDPL_BIOMOL_HPP
#define CDPL_BIOMOL_HPP

#include "CDPL/Biomol/PDBData.hpp"
#include "CDPL/Biomol/MMCIFData.hpp"
#include "CDPL/Biomol/AtomProperty.hpp"
#include "CDPL/Biomol/MolecularGraphProperty.hpp"
#include "CDPL/Biomol/AtomPropertyDefault.hpp"
#include "CDPL/Biomol/MolecularGraphPropertyDefault.hpp"
#include "CDPL/Biomol/AtomPropertyFlag.hpp"
#include "CDPL/Biomol/PDBFormatVersion.hpp"
#include "CDPL/Biomol/ResidueType.hpp"
#include "CDPL/Biomol/ProcessingFlags.hpp"
#include "CDPL/Biomol/DataFormat.hpp"
#include "CDPL/Biomol/ControlParameter.hpp"
#include "CDPL/Biomol/ControlParameterDefault.hpp"
#include "CDPL/Biomol/ControlParameterFunctions.hpp"
#include "CDPL/Biomol/MMCIFDataProcessingFunction.hpp"
#include "CDPL/Biomol/PDBMoleculeInputHandler.hpp"
#include "CDPL/Biomol/PDBMolecularGraphOutputHandler.hpp"
#include "CDPL/Biomol/MMTFMoleculeInputHandler.hpp"
#include "CDPL/Biomol/MMTFMolecularGraphOutputHandler.hpp"
#include "CDPL/Biomol/MMCIFMoleculeInputHandler.hpp"
#include "CDPL/Biomol/MMCIFMolecularGraphOutputHandler.hpp"
#include "CDPL/Biomol/PDBMoleculeReader.hpp"
#include "CDPL/Biomol/PDBMolecularGraphWriter.hpp"
#include "CDPL/Biomol/MMTFMoleculeReader.hpp"
#include "CDPL/Biomol/MMTFMolecularGraphWriter.hpp"
#include "CDPL/Biomol/MMCIFMoleculeReader.hpp"
#include "CDPL/Biomol/MMCIFMolecularGraphWriter.hpp"
#include "CDPL/Biomol/HierarchyViewNode.hpp"
#include "CDPL/Biomol/HierarchyViewModel.hpp"
#include "CDPL/Biomol/HierarchyViewChain.hpp"
#include "CDPL/Biomol/HierarchyViewFragment.hpp"
#include "CDPL/Biomol/HierarchyView.hpp"
#include "CDPL/Biomol/ResidueList.hpp"
#include "CDPL/Biomol/ResidueDictionary.hpp"
#include "CDPL/Biomol/AtomFunctions.hpp"
#include "CDPL/Biomol/MolecularGraphFunctions.hpp"
#include "CDPL/Biomol/MoleculeFunctions.hpp"
#include "CDPL/Biomol/UtilityFunctions.hpp"
#include "CDPL/Biomol/PDBGZMoleculeInputHandler.hpp"
#include "CDPL/Biomol/PDBGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Biomol/MMTFGZMoleculeInputHandler.hpp"
#include "CDPL/Biomol/MMTFGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Biomol/MMCIFGZMoleculeInputHandler.hpp"
#include "CDPL/Biomol/MMCIFGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Biomol/PDBBZ2MoleculeInputHandler.hpp"
#include "CDPL/Biomol/PDBBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Biomol/MMTFBZ2MoleculeInputHandler.hpp"
#include "CDPL/Biomol/MMTFBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Biomol/MMCIFBZ2MoleculeInputHandler.hpp"
#include "CDPL/Biomol/MMCIFBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Biomol/PDBGZMoleculeReader.hpp"
#include "CDPL/Biomol/PDBGZMolecularGraphWriter.hpp"
#include "CDPL/Biomol/MMTFGZMoleculeReader.hpp"
#include "CDPL/Biomol/MMTFGZMolecularGraphWriter.hpp"
#include "CDPL/Biomol/MMCIFGZMoleculeReader.hpp"
#include "CDPL/Biomol/MMCIFGZMolecularGraphWriter.hpp"
#include "CDPL/Biomol/PDBBZ2MoleculeReader.hpp"
#include "CDPL/Biomol/PDBBZ2MolecularGraphWriter.hpp"
#include "CDPL/Biomol/MMTFBZ2MoleculeReader.hpp"
#include "CDPL/Biomol/MMTFBZ2MolecularGraphWriter.hpp"
#include "CDPL/Biomol/MMCIFBZ2MoleculeReader.hpp"
#include "CDPL/Biomol/MMCIFBZ2MolecularGraphWriter.hpp"

#endif // CDPL_BIOMOL_HPP
