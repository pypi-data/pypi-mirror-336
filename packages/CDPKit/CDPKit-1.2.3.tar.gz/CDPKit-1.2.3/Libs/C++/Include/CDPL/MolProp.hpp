/* 
 * MolProp.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::MolProp.
 */

#ifndef CDPL_MOLPROP_HPP
#define CDPL_MOLPROP_HPP

#include "CDPL/MolProp/AtomProperty.hpp"
#include "CDPL/MolProp/BondProperty.hpp"
#include "CDPL/MolProp/AtomPropertyDefault.hpp"
#include "CDPL/MolProp/MassComposition.hpp"
#include "CDPL/MolProp/ElementHistogram.hpp"
#include "CDPL/MolProp/CoordinationGeometry.hpp"
#include "CDPL/MolProp/HBondDonorAtomType.hpp"
#include "CDPL/MolProp/HBondAcceptorAtomType.hpp"
#include "CDPL/MolProp/AtomFunctions.hpp"
#include "CDPL/MolProp/BondFunctions.hpp"
#include "CDPL/MolProp/AtomContainerFunctions.hpp"
#include "CDPL/MolProp/BondContainerFunctions.hpp"
#include "CDPL/MolProp/MolecularGraphFunctions.hpp"
#include "CDPL/MolProp/TPSACalculator.hpp"
#include "CDPL/MolProp/AtomHydrophobicityCalculator.hpp"
#include "CDPL/MolProp/LogSCalculator.hpp"
#include "CDPL/MolProp/XLogPCalculator.hpp"
#include "CDPL/MolProp/PEOESigmaChargeCalculator.hpp"
#include "CDPL/MolProp/MHMOPiChargeCalculator.hpp"
#include "CDPL/MolProp/HBondDonorAtomTyper.hpp"
#include "CDPL/MolProp/HBondAcceptorAtomTyper.hpp"

#endif // CDPL_MOLPROP_HPP
