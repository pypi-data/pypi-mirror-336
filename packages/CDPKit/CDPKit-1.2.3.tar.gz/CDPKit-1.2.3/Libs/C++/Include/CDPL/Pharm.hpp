/* 
 * Pharm.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::Pharm.
 */

#ifndef CDPL_PHARM_HPP
#define CDPL_PHARM_HPP

#include "CDPL/Pharm/Feature.hpp"
#include "CDPL/Pharm/BasicFeature.hpp"
#include "CDPL/Pharm/FeatureContainer.hpp"
#include "CDPL/Pharm/FeatureSet.hpp"
#include "CDPL/Pharm/Pharmacophore.hpp"
#include "CDPL/Pharm/BasicPharmacophore.hpp"
#include "CDPL/Pharm/FeatureMapping.hpp"
#include "CDPL/Pharm/FeatureTypeHistogram.hpp"
#include "CDPL/Pharm/Feature3DCoordinatesFunction.hpp"
#include "CDPL/Pharm/FeatureContainerProperty.hpp"
#include "CDPL/Pharm/FeatureProperty.hpp"
#include "CDPL/Pharm/FeatureContainerPropertyDefault.hpp"
#include "CDPL/Pharm/FeaturePropertyDefault.hpp"
#include "CDPL/Pharm/ControlParameter.hpp"
#include "CDPL/Pharm/ControlParameterDefault.hpp"
#include "CDPL/Pharm/DataFormat.hpp"
#include "CDPL/Pharm/FeatureType.hpp"
#include "CDPL/Pharm/FeatureGeometry.hpp"
#include "CDPL/Pharm/FeatureGenerator.hpp"
#include "CDPL/Pharm/PatternBasedFeatureGenerator.hpp"
#include "CDPL/Pharm/AromaticFeatureGenerator.hpp"
#include "CDPL/Pharm/HydrophobicFeatureGenerator.hpp"
#include "CDPL/Pharm/HydrophobicAtomFeatureGenerator.hpp"
#include "CDPL/Pharm/HBondAcceptorFeatureGenerator.hpp"
#include "CDPL/Pharm/HBondDonorFeatureGenerator.hpp"
#include "CDPL/Pharm/PosIonizableFeatureGenerator.hpp"
#include "CDPL/Pharm/NegIonizableFeatureGenerator.hpp"
#include "CDPL/Pharm/XBondAcceptorFeatureGenerator.hpp"
#include "CDPL/Pharm/XBondDonorFeatureGenerator.hpp"
#include "CDPL/Pharm/PharmacophoreGenerator.hpp"
#include "CDPL/Pharm/DefaultPharmacophoreGenerator.hpp"
#include "CDPL/Pharm/InteractionAnalyzer.hpp"
#include "CDPL/Pharm/DefaultInteractionAnalyzer.hpp"
#include "CDPL/Pharm/InteractionConstraintConnector.hpp"
#include "CDPL/Pharm/FeatureDistanceConstraint.hpp"
#include "CDPL/Pharm/HydrophobicInteractionConstraint.hpp"
#include "CDPL/Pharm/IonicInteractionConstraint.hpp"
#include "CDPL/Pharm/OrthogonalPiPiInteractionConstraint.hpp"
#include "CDPL/Pharm/ParallelPiPiInteractionConstraint.hpp"
#include "CDPL/Pharm/CationPiInteractionConstraint.hpp"
#include "CDPL/Pharm/HBondingInteractionConstraint.hpp"
#include "CDPL/Pharm/XBondingInteractionConstraint.hpp"
#include "CDPL/Pharm/InteractionPharmacophoreGenerator.hpp"
#include "CDPL/Pharm/FeatureInteractionScore.hpp"
#include "CDPL/Pharm/FeatureInteractionScoreCombiner.hpp"
#include "CDPL/Pharm/FeatureDistanceScore.hpp"
#include "CDPL/Pharm/HydrophobicInteractionScore.hpp"
#include "CDPL/Pharm/IonicInteractionScore.hpp"
#include "CDPL/Pharm/HBondingInteractionScore.hpp"
#include "CDPL/Pharm/XBondingInteractionScore.hpp"
#include "CDPL/Pharm/OrthogonalPiPiInteractionScore.hpp"
#include "CDPL/Pharm/ParallelPiPiInteractionScore.hpp"
#include "CDPL/Pharm/CationPiInteractionScore.hpp"
#include "CDPL/Pharm/SpatialFeatureMapping.hpp"
#include "CDPL/Pharm/PharmacophoreAlignment.hpp"
#include "CDPL/Pharm/FeatureTypeMatchFunctor.hpp"
#include "CDPL/Pharm/FeaturePairDistanceMatchFunctor.hpp"
#include "CDPL/Pharm/FeaturePositionMatchFunctor.hpp"
#include "CDPL/Pharm/FeatureGeometryMatchFunctor.hpp"
#include "CDPL/Pharm/PharmacophoreFitScore.hpp"
#include "CDPL/Pharm/CDFPharmacophoreInputHandler.hpp"
#include "CDPL/Pharm/PMLPharmacophoreInputHandler.hpp"
#include "CDPL/Pharm/CDFFeatureContainerOutputHandler.hpp"
#include "CDPL/Pharm/PMLFeatureContainerOutputHandler.hpp"
#include "CDPL/Pharm/CDFPharmacophoreReader.hpp"
#include "CDPL/Pharm/PMLPharmacophoreReader.hpp"
#include "CDPL/Pharm/CDFFeatureContainerWriter.hpp"
#include "CDPL/Pharm/PMLFeatureContainerWriter.hpp"
#include "CDPL/Pharm/ScreeningDBCreator.hpp"
#include "CDPL/Pharm/ScreeningDBAccessor.hpp"
#include "CDPL/Pharm/ScreeningProcessor.hpp"
#include "CDPL/Pharm/PharmacophoreFitScreeningScore.hpp"
#include "CDPL/Pharm/MoleculeFunctions.hpp"
#include "CDPL/Pharm/PharmacophoreFunctions.hpp"
#include "CDPL/Pharm/FeatureFunctions.hpp"
#include "CDPL/Pharm/FeatureContainerFunctions.hpp"
#include "CDPL/Pharm/FeatureSetFunctions.hpp"
#include "CDPL/Pharm/ControlParameterFunctions.hpp"
#include "CDPL/Pharm/UtilityFunctions.hpp"
#include "CDPL/Pharm/FileScreeningHitCollector.hpp"
#include "CDPL/Pharm/CDFGZPharmacophoreInputHandler.hpp"
#include "CDPL/Pharm/CDFBZ2PharmacophoreInputHandler.hpp"
#include "CDPL/Pharm/CDFGZFeatureContainerOutputHandler.hpp"
#include "CDPL/Pharm/CDFBZ2FeatureContainerOutputHandler.hpp"
#include "CDPL/Pharm/CDFGZPharmacophoreReader.hpp"
#include "CDPL/Pharm/CDFBZ2PharmacophoreReader.hpp"
#include "CDPL/Pharm/CDFGZFeatureContainerWriter.hpp"
#include "CDPL/Pharm/CDFBZ2FeatureContainerWriter.hpp"
#include "CDPL/Pharm/PharmacophoreReader.hpp"
#include "CDPL/Pharm/FeatureContainerWriter.hpp"
# include "CDPL/Pharm/PSDPharmacophoreInputHandler.hpp"
# include "CDPL/Pharm/PSDMoleculeInputHandler.hpp"
# include "CDPL/Pharm/PSDMolecularGraphOutputHandler.hpp"
# include "CDPL/Pharm/PSDPharmacophoreReader.hpp"
# include "CDPL/Pharm/PSDMoleculeReader.hpp"
# include "CDPL/Pharm/PSDMolecularGraphWriter.hpp"
# include "CDPL/Pharm/PSDScreeningDBCreator.hpp"
# include "CDPL/Pharm/PSDScreeningDBAccessor.hpp"

#endif // CDPL_PHARM_HPP
