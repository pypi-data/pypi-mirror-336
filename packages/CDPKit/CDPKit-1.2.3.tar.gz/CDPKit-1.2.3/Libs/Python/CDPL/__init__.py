##
# __init__.py  
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


from __future__ import absolute_import

import os
import sys
import platform

if sys.version_info[1] >= 8 and platform.system() == 'Windows':
    if 'site-packages' in __file__:
        os.add_dll_directory(os.path.dirname(__file__))
    else:
        os.add_dll_directory(os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), '@CDPKIT_EXECUTABLE_INSTALL_DIR@'))

from ._cdpl import *

__version__ = VERSION_STRING
