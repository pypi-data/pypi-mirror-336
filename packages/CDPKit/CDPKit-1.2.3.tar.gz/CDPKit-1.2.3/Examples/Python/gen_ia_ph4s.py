#!/bin/env python

##
# gen_ia_ph4s.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys
import argparse

import CDPL.Chem as Chem
import CDPL.Biomol as Biomol
import CDPL.Pharm as Pharm
import CDPL.MolProp as MolProp


# reads and preprocesses the specified receptor structure
def readAndPrepareReceptorStructure(args: argparse.Namespace) -> Chem.Molecule:
    # create reader for receptor structure (format specified by file extension)
    reader = Chem.MoleculeReader(args.receptor_file) 
    
    sup_fmts = [ Chem.DataFormat.MOL2,
                 Biomol.DataFormat.PDB,
                 Biomol.DataFormat.MMTF,
                 Biomol.DataFormat.MMCIF ]
                        
    if reader.getDataFormat() not in sup_fmts:   # check if the format is supported by this script
        sys.exit('Error: receptor input file format not supported')

    rec_mol = Chem.BasicMolecule()    # create an instance of the default implementation of the
                                      # Chem.Molecule interface that will store the receptor struct.
    try:
        if not reader.read(rec_mol):  # read receptor structure
            sys.exit('Error: reading receptor structure failed')

    except Exception as e:
        sys.exit('Error: reading receptor structure failed:\n' + str(e))            

    # preprocess the receptor structure (removal of residues and
    # calculation of properties required by the pharm. generation procedure)
    try:
        rem_atoms = False

        # delete atoms belonging to residues that should be stripped
        if args.strip_res_list:            
            atoms_to_rem = Chem.Fragment() # will store the atoms to delete
            res_to_strip = { tlc.upper() for tlc in args.strip_res_list }
        
            for atom in rec_mol.atoms:     # identify and note atoms belonging to the stripped residues
                if Biomol.getResidueCode(atom).upper() in res_to_strip:
                    atoms_to_rem.addAtom(atom)

            if atoms_to_rem.numAtoms > 0:
                rec_mol -= atoms_to_rem    # delete atoms from the receptor structure
                rem_atoms = True

                if not args.quiet:
                    print('! Removed %s atoms from the receptor structure' % str(atoms_to_rem.numAtoms))

        # prepares the receptor structure for pharmacophore generation
        Chem.perceiveSSSR(rec_mol, rem_atoms)
        Chem.setRingFlags(rec_mol, rem_atoms)
        Chem.calcImplicitHydrogenCounts(rec_mol, rem_atoms);
        Chem.perceiveHybridizationStates(rec_mol, rem_atoms);
        Chem.setAromaticityFlags(rec_mol, rem_atoms);

        if Chem.makeHydrogenComplete(rec_mol):                    # make implicit hydrogens (if any) explicit
            Chem.calcHydrogen3DCoordinates(rec_mol)               # calculate 3D coordinates for the added expl. hydrogens
            Biomol.setHydrogenResidueSequenceInfo(rec_mol, False) # set residue information for the added expl. hydrogens

        MolProp.calcAtomHydrophobicities(rec_mol, False)          # calculate atom hydrophobicity values (needed for hydrophobic
                                                                  # pharm. feature generation)
    except Exception as e:
        sys.exit('Error: processing of receptor structure failed: ' + str(e))            

    return rec_mol
   
def parseArgs() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description='Generates pharmacophores describing the interactions between a given receptor structure and a set of ligand molecules.')

    parser.add_argument('-r',
                        dest='receptor_file',
                        required=True,
                        metavar='<file>',
                        help='Receptor structure input file (*.mol2, *.pdb, *.mmtf, *.cif, *.mmcif)')
    parser.add_argument('-l',
                        dest='ligands_file',
                        required=True,
                        metavar='<file>',
                        help='Ligand structure input file (*.sdf, *.mol2, *.cdf)')
    parser.add_argument('-o',
                        dest='out_file',
                        required=True,
                        metavar='<file>',
                        help='Pharmacophore output file (*.pml, *.cdf)')
    parser.add_argument('-s',
                        dest='strip_res_list',
                        required=False,
                        metavar='<three-letter code>',
                        nargs='+',
                        help='Whitespace separated list of PDB three-letter codes specifying residues to remove from the receptor structure (e.g. an existing ligand)')
    parser.add_argument('-q',
                        dest='quiet',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Disable progress output (default: false)')
    parser.add_argument('-x',
                        dest='gen_x_vols',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Generate exclusion volumes (default: false)')

    return parser.parse_args()

def main() -> None:
    args = parseArgs()

    rec_mol = readAndPrepareReceptorStructure(args)          # read and preprocess the receptor structure
    lig_reader = Chem.MoleculeReader(args.ligands_file)      # create reader for the ligand input file (format specified by file extension)
    ph4_writer = Pharm.FeatureContainerWriter(args.out_file) # create writer for the generated pharmacophores (format specified by file extension)
 
    lig_mol = Chem.BasicMolecule()          # create an instance of the default implementation of the
                                            # Chem.Molecule interface that will store the ligand structures
    ia_ph4 = Pharm.BasicPharmacophore()     # create an instance of the default implementation of the Pharm.Pharmacophore
                                            # interface that will store the generated pharmacophores

    ph4_gen = Pharm.InteractionPharmacophoreGenerator() # create an instance of the pharmacophore generator

    ph4_gen.addExclusionVolumes(args.gen_x_vols)        # specify whether to generate exclusion volume spheres 
                                                        # on pharm. feature atoms of interacting residues
    try:
        i = 1

        # read and process ligand molecules one after the other until the end of input has been reached (or a severe error occurs)
        while lig_reader.read(lig_mol):
            mol_id = Chem.getName(lig_mol).strip() # compose a simple ligand identifier for messages

            if mol_id == '':
                mol_id = '#' + str(i)  # fallback if name is empty or not available
            else:
                mol_id = '\'%s\' (#%s)' % (mol_id, str(i))

            if not args.quiet:
                print('- Generating interaction pharmacophore of molecule %s...' % mol_id)

            try:
                Pharm.prepareForPharmacophoreGeneration(lig_mol) # make ligand ready for pharm. generation

                ph4_gen.generate(lig_mol, rec_mol, ia_ph4, True) # generate the pharmacophore (True = extract ligand environment residues on-the-fly)

                if not args.quiet:
                     print(' -> Generated %s features: %s' % (str(ia_ph4.numFeatures), Pharm.generateFeatureTypeHistogramString(ia_ph4)))
                
                try:
                    if not ph4_writer.write(ia_ph4): # output pharmacophore
                        sys.exit('Error: writing interaction pharmacophore of molecule %s failed' % mol_id)

                except Exception as e:               # handle exception raised in case of severe write errors
                    sys.exit('Error: writing interaction pharmacophore of molecule %s failed: %s' % (mol_id, str(e)))
                
            except Exception as e:                   # handle exception raised in case of severe processing errors
                sys.exit('Error: interaction pharmacophore generation for molecule %s failed: %s' % (mol_id, str(e)))

            i += 1
            
    except Exception as e:                           # handle exception raised in case of severe read errors
        sys.exit('Error: reading molecule %s failed: %s' % (str(i), str(e)))

    if not args.quiet:
        print('Done!')

    ph4_writer.close()
    sys.exit(0)
        
if __name__ == '__main__':
    main()
