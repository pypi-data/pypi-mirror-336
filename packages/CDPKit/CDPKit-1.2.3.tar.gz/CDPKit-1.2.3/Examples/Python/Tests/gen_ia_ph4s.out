! Removed 27 atoms from the receptor structure
- Generating interaction pharmacophore of molecule 'LS2' (#1)...
 -> Generated 27 features: H(3), HBD(3), HBA(3), XV(18)
Done!
